docs: Added docstring for `COO.mT`

DeaMariaLeon opened the pull request:
Could you please tell me if the docstrings should be like this?

hameerabbasi commented on mkdocs.yml:
```diff
@@ -59,6 +59,7 @@ plugins:
               show_if_no_docstring: true
               members_order: source
               docstring_style: numpy
+              show_source: false
```
```suggestion
```

I'd keep the source visible

hameerabbasi commented on docs/operations.md:
```diff
@@ -77,7 +77,7 @@ In these cases, they will produce an output with a fill value of `1` or `True`,
 
 If densification is needed, it must be explicit. In other words, you must call
-[`sparse.SparseArray.todense`][] on the [`sparse.SparseArray`][] object. If both operands are [sparse.SparseArray][],
+[`sparse.SparseArray.todense`][] on the [`sparse.SparseArray`][] object. If both operands are [`sparse.SparseArray`][],
```
Thanks for the fix.

hameerabbasi commented on sparse/numba_backend/_coo/core.py:
```diff
@@ -851,6 +851,38 @@ def T(self):
+
+        Examples
+        --------
+
```
```suggestion
```

Unless the newline is needed for proper rendering, I'd remove it.

DeaMariaLeon replied:
Done.

hameerabbasi reviewed (changes requested):
Some minor nits, but everything else LGTM.

One comment: I'd copy the descriptions from the [Array API docs](https://data-apis.org/array-api/latest/) as much as I can.

DeaMariaLeon commented:
>I'd copy the descriptions from the [Array API docs](https://data-apis.org/array-api/latest/) as much as I can.

I changed the description then. 

I think I addressed the rest of the comments too. Please let me know if something else is needed.

hameerabbasi commented:
Thanks for the work on this @DeaMariaLeon! This should merge once CI passes.
